File: coding/python/stackusingarray.py

class stackusingarray:

    # Constructor
    def __init__(self, size):

        self.capacity = size
        self.stack = [0] * size
        self.top = -1


    #---------------------------
    # Pusk Operation
    #---------------------------

    def push(self, value):

        if self.is_full():
            print("Stack Overflow....")
            return

        self.top += 1
        self.stack[self.top] = value
        print(f"{value} pushed into stack")

    #-----------------------------
    # Pop Operation
    #-------------------------------

    def pop(self):

        if self.is_empty():
            print("STack Underflow .....")
            return None

        value = self.stack[self.top]
        self.top -= 1

        return value

    #-------------------------------
    # Peep Operation
    #-------------------------------

    def peek(self):

        if self.is_empty():
            print("Stack is Empty ...222 !!!!!!!")
            return None

        return self.stack[self.top]

    #-------------------------
    # Check Empty
    #-------------------------

    def is_empty(self):

        return self.top == -1

    #----------------------------
    #Check Full
    #----------------------------

    def is_full(self):

        return self.top == self.capacity - 1

    #------------------------------
    # Display Stack
    #--------------------------------

    def display(self):

        if self.is_empty():
            print("Stack is Empty 333 !!!!")
            return

        print("Inside Display ..... Stack Elements:")

        for i in range(self.top, -1, -1):
            print(self.stack[i])


#--------------------
# Main Program
#------------------------

stack = stackusingarray(5)

stack.push(100)
stack.push(200)
stack.push(300)

stack.display()

print("Top Element : ", stack.peek())
print("Popped Element: ", stack.pop())

stack.display()

File: coding/java/stackUsingArray.java
package coding.java;

public class stackUsingArray {
    private int array[];
    private int top;
    private int capacity;

    stackUsingArray(int capacity) {
        this.array = new int[capacity];
        this.capacity = capacity;
        this.top = -1;
    }

    public void push(int item){
        if (isFull()){
            throw new RuntimeException("Stack is Full ..1111 !!!");
        }
        array[++top] = item;
    }

    public int pop(){
        if (isEmpty()) {
            throw new RuntimeException("Stack is empty ...2222 !!!");
        }
        return array[top--];
    }

    public int peek(){
        if (isEmpty()){
            throw new RuntimeException("Stack is empty ...333 !!!");
        }
        return array[top];
    }

    public boolean isEmpty(){
        return top == -1;
    }

    public boolean isFull(){
        return top == capacity - 1;
    }

    public void display(){

        if(isEmpty()){
            System.out.println("Stack is Empty !!!!!");
            return;
        }

        System.out.println("Stack Elements ....");

        for(int i = top; i >= 0; i--){
            System.out.println(array[i]);
        }
    }

    public static void main(String[] args) {
        
        stackUsingArray stack = new stackUsingArray(5);

        stack.push(10);
        stack.push(20);
        stack.push(30);
        stack.push(40);
        stack.push(50);
        stack.push(60);
        

        stack.display();

        System.out.println("Top Element : " + stack.peek());
        System.out.println("Pooped Element : " + stack.pop());

        stack.display();
    }
    
}
